### PR TITLE
SANDBOX-677 use go-version-file

### PR DESCRIPTION
**Pull request by ranakan19**
uses go-version-file instead of go-version. One less place to change when making updates.
Changes order in github action to first checkout code and then install go.

related PRs:
https://github.com/codeready-toolchain/api/pull/434
https://github.com/codeready-toolchain/toolchain-common/pull/415
https://github.com/codeready-toolchain/member-operator/pull/584
https://github.com/codeready-toolchain/registration-service/pull/445
https://github.com/codeready-toolchain/host-operator/pull/1060
https://github.com/codeready-toolchain/toolchain-e2e/pull/1014

### CODE REVIEW

**Review by alexeykazakov — APPROVED** [review]

**Review by fbm3307 — APPROVED** [review]
Great idea . lgtm

File: .github/workflows/user-identity-mapper.yml [hunk 1]
```diff
@@ -19,16 +19,16 @@ jobs:
```
Comment (fbm3307) [inline .github/workflows/user-identity-mapper.yml:19]: Can you remove GO_VERSION: from here too , it is not used anymore i guess

**Review by fbm3307 — COMMENTED** [review]
Thanks for the PR

File: .github/workflows/user-identity-mapper.yml (hunk 1 above)
Comment (ranakan19) [reply .github/workflows/user-identity-mapper.yml:18]: whoops, good eye, I somehow missed this - thought i had removed it.
removed in 6950e0f

**Review by MatousJobanek — APPROVED** [review]

**Pull request merged**